jack-core/src/com/rapleaf/jack/transaction/DbPoolManager.java

```diff
@@ -110,6 +120,33 @@ public void close() {
     connectionPool.close();
   }
 
+  @Override
+  public DbMetrics getMetrics() {
+    if (metricsTrackingEnabled) {
+      metrics.update(false, connectionPool);
+      return metrics;
+    } else {
+      LOG.info("logging disabled, db metrics cannot be accessed");
+      return null;
+    }
```

Comment:
Another way is to create a second implementation of the metrics interface which does nothing. When tracking is disabled, assign the `metrics` variable to this implementation. But we probably don't need it for now.